### ISSUE TITLE
Sayma hardware testing checklist

### ISSUE THREAD

**Issue by jbqubit** [m-labs/sinara#224]
# Key Dates
__5/29__:  2 of Sayma_RTM delivered to WUT
__6/8__: 2 of Sayma_RTM delivered to WUT
__6/25__: 1 Sayma_AMC shipped to M-Labs (Florent)
    - ​​with [RTM loopback](https://github.com/m-labs/sinara/blob/master/ARTIQ_ALTIUM/PCB_RTM_loopback/PCB_RTM_Loopback.PDF) board with clock oscillators to test GTH and IOs
    - with [SATA adapter](https://github.com/m-labs/sinara/blob/master/ARTIQ_ALTIUM/PCB_AMC_SATA/AMC_SATA_Adapter.PDF)

# AMC
AMC errata tracked in #209. 
- [x] power supplies, sequencing 
- [ ] AMC+RTM power dissipation under load in chassis 
- [x] oscillators 
- [x] MMC 
- [x] MMC update over USB 
- [ ] MMC ethernet
- [x] FPGA UARTs 
- [x] FPGA programming 
- [x] JTAG over USB
- [x] LEDs
- [x] SMA outputs 
- [x] AMC Ethernet connectvity 
- [x] SDRAM 
- [x] GTH transceivers and SFP 
- [x] Silabs clock chip 
- [x] SFP
- [x] FLASH 
- [x] I2C tree (only AMC side)
- [x] FMC
- [x] AMC to RTM FPGA bridge

# RTM
RTM errata tracked in #216. 
- [x] oscillators 
- [x] basic clock distribution
- [x] FPGA UART 
- [x] FPGA programming 
- [x] JTAG over USB
- [x] I2C tree excluding analog front end (AFE) mezzanines 
- [x] I2C tree including AFE
- [x] AFE 
- [x] SPI
- [ ] ADC
- [ ] DAC

**Comment by jbqubit** [comment]
@gkasprow I've had several inquires as to the status of Sayma hardware testing. Here's a checklist for people to follow who want to track progress.

**Comment by gkasprow** [comment]
The SDRAM 32 works perfectly
![obraz](https://user-images.githubusercontent.com/4325054/27405421-c1a5c288-56d1-11e7-9fed-cfc3311038f3.png)

**Comment by jbqubit** [comment]
Shift date for Sayma shipment to M-Labs. Now tomorrow.  

> 6/22: 1 each of Sayma_RTM & Sayma_AMC shipped to M-Labs (Florent)

**Comment by gkasprow** [comment]
Second DDR 64 controller working. I really hate Vivado! It took 2 days to synthesize and make running Xilinx example design...
![obraz](https://user-images.githubusercontent.com/4325054/27509203-db947360-58f7-11e7-9331-7d4c857324f5.png)

**Comment by gkasprow** [comment]
16 GT links running in parallel at 12Gbit. Chip temperature 40deg without heatsink and with air cooling. After a few minutes of work no errors detected.
![obraz](https://user-images.githubusercontent.com/4325054/27509458-90c486da-58fd-11e7-94e4-ed8d76be2701.png)

**Comment by gkasprow** [comment]
Example eye scan using IBERT tool. All tests done with RTM loopback that mimics poor matching.
![obraz](https://user-images.githubusercontent.com/4325054/27509625-8c85f582-5901-11e7-997d-463df9ae2974.png)

**Comment by gkasprow** [comment]
SFP to SFP connectivity with 30cm copper cable
![Uploading obraz.png…]()

**Comment by gkasprow** [comment]
No troubles with PRBS31 as well
![obraz](https://user-images.githubusercontent.com/4325054/27509796-d586a92c-5904-11e7-8663-27f41b162d5a.png)

**Comment by jordens** [comment]
Just as a sidenote: That eye doesn't look physical to me. It is also unlike anything from #166. There are features much shorter than 0.01 UI, i.e. 1 ps in there. I'd suspect some Vivado bug in there.

**Comment by gkasprow** [comment]
@jordens This has very low resolution - 8 units vertical, 8 units horizontal. In ISE it looked better.

**Comment by gkasprow** [comment]
I tested Sayma with FMC->VHDCI adapter -> VHDCI carrier -> 3U RJ45 in a loopback. Works fine.
The signal enters via SMA, passes 32 times through FMC and 3U RJ45 extender and goes to the SMA output. Below is screenshot.
![obraz](https://user-images.githubusercontent.com/4325054/27635588-edaf2e1a-5c07-11e7-8d01-0b6abb9eb881.png)

**Comment by gkasprow** [comment]
SPI FLASHx4  works fine. FPGA boots after ~one second.

![obraz](https://user-images.githubusercontent.com/4325054/27648159-ee6557fa-5c2d-11e7-98ac-7dea3401fa27.png)

**Comment by gkasprow** [comment]
There was issue with SCANSTA chip. I didn't work.  I figured out that switching it to mode with master port B1 (power on with JTAG probe disconnected) and then pulling TRSTB to GND made it working with master port B0. It seems that it needs some time after power ON. And indeed, configuration happen too early during power on. 200ms delay in MMC software solved that issue,
There was the same issue with Silabs was that it needs some time after power on. I2C initialisation failed when started too early and locked the chip. Initial delay made it working.

**Comment by gkasprow** [comment]
JTAG on RTM works
![obraz](https://user-images.githubusercontent.com/4325054/27793712-663aaaa8-5fff-11e7-85c1-5794cd8f477b.png)

**Comment by gkasprow** [comment]
Silabs on RTM is also working. I set it to free-run at 125MHz and will test ethernet connectivity over GTP and SATA cable.

**Comment by gkasprow** [comment]
Status update:
Mux tree works fine.
I managed to communicate with PLL chip from little FORTH program that does bit-bang SPI. I discovered that PLL chip responds with right ID but logic levels are compromised. The HMC7043 does not release SPI bus for some reason so I will have to either add 3-state buffer or route dedicated line to FPGA.
![tek00045](https://user-images.githubusercontent.com/4325054/27967016-b668e0d6-6341-11e7-8f7f-56f1942a429c.png)

**Comment by hartytp** [comment]
@gkasprow Great! Did you get RF out of the HMC830?

**Comment by sbourdeauducq** [comment]
What does Analog Devices tech support have to say about the HMC7043 not releasing the SPI bus? Maybe it just needs some clock pulses with CS deasserted?

**Comment by gkasprow** [comment]
This pin is bidir, and I keep CS high.

Maybe there is something else that block this line…

Clock line is also common for both chips.

**Comment by gkasprow** [comment]
@hartytp Not yet, I was fighting with SPI...

**Comment by jbqubit** [comment]
@gkasprow I added "SPI" checkbox for RTM and checked it since you reported it working in #216. Nice!

**Comment by gkasprow** [comment]
@jbqubit One SPI chip works.
I'm trying to make others running and it is tricky.

**Comment by gkasprow** [comment]
RTM clock switching, PLL, distribution works. This SPI is really tricky and needs correct init sequence.

**Comment by gkasprow** [comment]
ADC and DAC SPI works
![obraz](https://user-images.githubusercontent.com/4325054/28125914-51780d42-6728-11e7-83ad-e62aea1cdd7a.png)

**Comment by gkasprow** [comment]
Bizzare I2C behaviour.
The TCA9539 needs to have any write operation performed first to respond to read operations later on.
All other chips ACK read operations.
once you make write operation, the TCA9539 responds later on to any read operations.
If you don't to write operation first, the chip does not ACK its own address...

**Comment by hartytp** [comment]
Great! So, it's just the DACs and ADCs left to test then...

**Comment by gkasprow** [comment]
power supply issue is known - it was caused by too rapid switching of RTM supply - main 12V rail went to 6V for a moment and then caused 1.8V overvoltage event. I added R=1.5k C=1uF to the PMOS switch to slow down RTM 12V slope and it works fine. This was already added as an issue by Kermit - he noticed it on my previous AMC board.

**Comment by gkasprow** [comment]
@sbourdeauducq  @enjoy-digital @jordens @jbqubit AMC+RTM are tested and working, I'm not aware of any issues. I didn't test:
- USB JTAG
- JESD204B to DAC and ADC
- Artix FPGA programming by Kintex.
- MMC access to Eternet
 I'm ready to send it to France.
Tomorrow morning I will send it.
On Monday I will receive 8 AMCs. I will pass one AMC+RTM set to Creotech, they will test ADC JESD204B and remaining less critical parts. I will focus on 3U boards now.

**Comment by hartytp** [comment]
@gkasprow Massive congratulations to you and your team for designing and testing those boards!

**Comment by gkasprow** [comment]
Thanks:)

**Comment by gkasprow** [comment]
8 Sayma AMC are produced. On Friday I will test them.

**Comment by gkasprow** [comment]
Front panels for Sayma AMC and RTM finally arrived. I will take them with Saymas to NACTI.

**Comment by jbqubit** [comment]
​Excellent! ​

**Comment by hartytp** [comment]
This issue covered basic functionality testing, which is complete.

Let's move to a new issue for more detailed analog testing (phase noise etc).
